eclair-core/src/test/scala/fr/acinq/eclair/TestDatabases.scala

```diff
@@ -112,13 +115,34 @@ object TestDatabases {
     implicit val system: ActorSystem = ActorSystem()
 
     // @formatter:off
-    override val connection: PgConnection = pg.getPostgresDatabase.getConnection.asInstanceOf[PgConnection]
+    override val connection: PgConnection = datasource.getConnection.asInstanceOf[PgConnection]
     // NB: we use a lazy val here: databases won't be initialized until we reference that variable
     override lazy val db: Databases = Databases.PostgresDatabases(hikariConfig, UUID.randomUUID(), lock, jdbcUrlFile_opt = Some(jdbcUrlFile), readOnlyUser_opt = None, resetJsonColumns = false, safetyChecks_opt = None)
-    override def close(): Unit = pg.close()
+    override def close(): Unit = {
+      db.asInstanceOf[PostgresDatabases].dataSource.close()
+      connection.close()
+      system.terminate()
+    }
     // @formatter:on
   }
 
+  object TestPgDatabases {
+    /** single instance */
+    private val pg = EmbeddedPostgres.start()
+
+    def getNewDatabase: DataSource = {
```

Comment:
nit: should look like a function call, not a getter on an internal field:
```suggestion
    def getNewDatabase(): DataSource = {
```